packages/ui/src/modals/AddMembershipModal/AddMembershipModal.tsx

```diff
@@ -38,12 +41,29 @@ export const AddMembershipModal = ({ onClose }: MembershipModalProps) => {
     [JSON.stringify(transactionParams)]
   )
 
-  const onDone = (result: boolean) => setStep(result ? 'SUCCESS' : 'ERROR')
-
   if (step === 'PREPARE' || !transactionParams) {
     return <MembershipFormModal onClose={onClose} onSubmit={onSubmit} membershipPrice={membershipPrice} />
   }
 
+  const onDone = (result: boolean, events: EventRecord[]) => {
+    const memberId = events.find((event) => event.event.method === 'MemberRegistered')?.event.data[0].toString()
```

Comment:
We have the created memberId so we can use it later in the `SUCESS` step.

This can be separate PR, though as it would require to:

- pass the `memberId` to the success modal
- use that `memberId` to invoke "view my profile" button

Reply:
Actually it deserves its own ticket – could you create one?

Tricky part to research: hide current modal and display arbitrary one.